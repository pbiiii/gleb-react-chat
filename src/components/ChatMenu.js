import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles/index';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import ChatType from '../types/ChatType';
import ActiveUserType from '../types/ActiveUserType';

const styles = theme => ({
    chatMenuContainer: {
        color: '#fff',
        display: 'flex',
        alignItems: 'center',
    },
    chatMenuTitle: {
        color: '#fff',
        marginLeft: theme.spacing.unit * 2,
    },
});

class ChatMenu extends React.Component {
    static propTypes = {
        classes: PropTypes.objectOf(PropTypes.string).isRequired,
        leaveChat: PropTypes.func.isRequired,
        deleteChat: PropTypes.func.isRequired,
        activeChat: PropTypes.shape(ChatType).isRequired,
        activeUser: PropTypes.shape(ActiveUserType).isRequired,
        disabled: PropTypes.bool.isRequired,
    };

    state = {
        anchorEl: null,
    };

    openMenu = event => this.setState({ anchorEl: event.currentTarget });

    closeMenu = () => {
        this.setState({ anchorEl: null });
    };

    render() {
        const {
            leaveChat, deleteChat, activeChat, activeUser, classes, disabled,
        } = this.props;
        const { anchorEl } = this.state;
        const chatTitle = activeChat.title.substring(0, 2).toUpperCase();
        return (
            <div className={classes.chatMenuContainer}>
                <Avatar>{chatTitle}</Avatar>
                <Typography variant="title" className={classes.chatMenuTitle}>
                    {activeChat.title}
                </Typography>
                {activeUser.isChatMember && (
                    <React.Fragment>
                        <IconButton
                            color="inherit"
                            aria-owns={anchorEl ? 'chat-menu' : null}
                            aria-haspopup="true"
                            disabled={disabled}
                            onClick={this.openMenu}
                        >
                            <MoreVertIcon />
                        </IconButton>
                        <Menu
                            id="chat-menu"
                            anchorEl={anchorEl}
                            open={Boolean(anchorEl)}
                            onClose={this.closeMenu}
                        >
                            {activeUser.isMember && (
                                <MenuItem onClick={leaveChat}>Leave chat</MenuItem>
                            )}
                            {activeUser.isCreator && (
                                <MenuItem onClick={deleteChat}>Delete chat</MenuItem>
                            )}
                        </Menu>
                    </React.Fragment>
                )}
            </div>
        );
    }
}

export default withStyles(styles)(ChatMenu);
