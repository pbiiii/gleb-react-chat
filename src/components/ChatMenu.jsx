import React from 'react';
import {withStyles} from "@material-ui/core/styles/index";
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import IconButton from '@material-ui/core/IconButton';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
    chatMenuContainer: {
        color: '#fff',
        display: 'flex',
        alignItems: 'center'
    },
    chatMenuTitle: {
        color: '#fff',
        marginLeft: theme.spacing.unit * 2
    }
});

class ChatMenuComponent extends React.Component {
    state = {
        anchorEl: null,
        modalIsOpen: false,
        editProfileForm: {
            userName: '',
            firstName: '',
            lastName: '',
        }
    };

    openMenu = event => {
        this.setState({ anchorEl: event.currentTarget });
    };

    closeMenu = () => {
        this.setState({ anchorEl: null });
    };

    render() {
        const {
            // disabled = false,
            activeUser,
            onLeaveClick,
            activeChat,
            classes,
        } = this.props
        const disabled = false
        const { anchorEl } = this.state;
        const chatTitle = activeChat.title.substring(0, 2).toUpperCase()
        return (
            <Typography variant="body1" className={classes.chatMenuContainer}>
                <Avatar>{chatTitle}</Avatar>
                <Typography variant="title" className={classes.chatMenuTitle}>
                    {activeChat.title}
                </Typography>
                <IconButton
                    color="inherit"
                    aria-owns={anchorEl ? 'chat-menu' : null}
                    aria-haspopup="true"
                    disabled={disabled}
                    onClick={this.openMenu}
                >
                    <MoreVertIcon/>
                </IconButton>
                <Menu
                    id="chat-menu"
                    anchorEl={anchorEl}
                    open={Boolean(anchorEl)}
                    onClose={this.closeMenu}
                >
                    <MenuItem
                        onClick={onLeaveClick}
                    >
                        Leave chat
                    </MenuItem>
                </Menu>
            </Typography>
        );
    }
}

export const ChatMenu = withStyles(styles)(ChatMenuComponent);