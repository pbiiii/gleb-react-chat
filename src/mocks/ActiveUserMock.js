import UserMock from './UserMock';

export default {
    ...UserMock,
    isMember: false,
    isCreator: true,
    isChatMember: true,
};
