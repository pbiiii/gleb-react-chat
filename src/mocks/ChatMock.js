import UserMock from './UserMock';

export default {
    _id: 'ashasdaksdmkl1mkl',
    creator: { ...UserMock, _id: 2 },
    title: 'Mock title',
};
