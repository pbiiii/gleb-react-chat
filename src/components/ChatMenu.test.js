/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMenu from './ChatMenu';
import ChatMock from '../mocks/ChatMock';
import ActiveUserMock from '../mocks/ActiveUserMock';

const mockProps = {
    activeChat: { ...ChatMock },
    activeUser: { ...ActiveUserMock },
    disabled: false,
    leaveChat: () => {},
    deleteChat: () => {},
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMenu {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<ChatMenu {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
