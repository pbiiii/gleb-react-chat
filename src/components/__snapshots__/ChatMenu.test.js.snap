// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="ChatMenu-chatMenuContainer-70"
>
  <div
    className="MuiAvatar-root-72 MuiAvatar-colorDefault-73"
  >
    MO
  </div>
  <h2
    className="MuiTypography-root-75 MuiTypography-title-81 ChatMenu-chatMenuTitle-71"
  >
    Mock title
  </h2>
  <button
    aria-haspopup="true"
    aria-owns={null}
    className="MuiButtonBase-root-116 MuiIconButton-root-110 MuiIconButton-colorInherit-111"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-115"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-119"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-132"
    />
  </button>
</div>
`;
